client-next/src/components/MapView/MapView.tsx

```diff
@@ -29,20 +33,33 @@ export function MapView({
   loading: boolean;
 }) {
   const onMapDoubleClick = useMapDoubleClick({ tripQueryVariables, setTripQueryVariables });
-  const [showPopup, setShowPopup] = useState<LngLat | null>(null);
+  const [showContextPopup, setShowContextPopup] = useState<LngLat | null>(null);
+  const [showPropsPopup, setShowPropsPopup] = useState<PopupData | null>(null);
 
   return (
     <div className="map-container below-content">
       <Map
         // @ts-ignore
         mapLib={import('maplibre-gl')}
         // @ts-ignore
-        mapStyle={mapStyle}
+        mapStyle={styleUrl}
         initialViewState={initialViewState}
         onDblClick={onMapDoubleClick}
         onContextMenu={(e) => {
-          setShowPopup(e.lngLat);
+          setShowContextPopup(e.lngLat);
         }}
+        interactiveLayerIds={['regular-stop']}
+        onClick={(e) => {
+          if (e.features) {
+            const props = e.features[0];
```

Comment:
A little context would help here. Why are we picking the first feature, and what is the significance of that feature's properties? The callback could even be extracted and named for clarity.

Reply:
The variable name `props` is also confusing. Should it not instead be `feature`?

Reply:
All of these things have been improved in c5dc28984a83a968660c2478c2a6b3301fa7a6d4